### PR TITLE
Implements NSSplitViewController (Yay!)

### PR DESCRIPTION
**Pull request by jleandroperez**
## Details:
In this PR we're **finally implementing NSSplitViewController.** 

- We're "stealing" the Tags / Notes / Editor Views from the hierarchy defined in **MainMenu.xib**.
- Endgame is to initialize all of our ViewControllers in Swift
- **NoteEditorViewController** is now in charge of rendering its own background
- We're deleting **so much dead / outdated code**!!!
- There's a **Rick Level Hack** to force **NSSplitView** redraw its dividers, required in OSX < 10.15

Ladies and gentlemen, may I bug you with the **coolest macOS PR ever?** 
**(I mean it!!!)**

cc @aerych & @emilylaguna 

Ref. #458

## ☠️ Known Issues ☠️
[NSScrollView may scroll during resize](https://github.com/Automattic/simplenote-macos/issues/536). This problem is present in `develop`, but may be extra noticeable, since now the SplitView has animations. **To be fixed really soon!**

### macOS Catalina | @aerych 👇👇👇 (Thank you sir!!!)
- [x] Verify switching Dark / Light mode causes the Split Dividers to render in the right color
- [x] Verify the Editor's Background color **always** looks great (Light / Dark)
- [x] Verify that toggling the **Sidebar** (CMD + T) properly shows / hides the Tags List panel
- [x] Verify that toggling **Focus Mode** (Shift + CMD + F) properly shows / hides everything
- [x] Verify that the Tags List can be resized
- [x] Verify that the Notes List can be resized

### macOS High Sierra | @emilylaguna  👇👇👇 (Thank you Em!!!)
- [ ] Verify switching Dark / Light mode causes the Split Dividers to render with the right color
- [ ] Verify the Editor's Background color **always** looks great (Light / Dark)
- [ ] Verify that toggling the **Sidebar** (CMD + T) properly shows / hides the Tags List panel
- [ ] Verify that toggling **Focus Mode** (Shift + CMD + F) properly shows / hides everything
- [ ] Verify that the Tags List can be resized
- [ ] Verify that the Notes List can be resized

### Fresh Install | @aerych 👇👇👇 (Thank you sir!!)
1. Run `defaults delete com.automattic.SimplenoteMac.beta` in your console
2. Log into Simplenote!

- [x] Verify that Simplenote's Tag List and Notes List look great

**Note:** 
Fresh installs, in `develop`, are launching the app with invalid SplitView panel sizes. Not anymore!
## 🗒️ Release 🗒️
`RELEASE-NOTES.txt` was updated in 4f34c11 with:

> Sidebar and Notes List will now be presented with a super awesome animation

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 ------
 - New Color Scheme for Light and Dark Modes!
 - Simplenote will now react dynamically to macOS Appearance updates when the System Theme is selected.
+- Sidebar and Notes List will now be presented with a super awesome animation
```
Comment (aerych) [inline RELEASE-NOTES.txt:5]: Maybe add a period `.` at the end

File: Simplenote/SimplenoteAppDelegate+Swift.swift [hunk 2]
```diff
@@ -1,6 +1,46 @@
 import Foundation
 
 
+// MARK: - Initialization
+//
+extension SimplenoteAppDelegate {
+
+    @objc
+    func configureSplitView() {
+        precondition(tagListViewController != nil)
```
Comment (aerych) [inline Simplenote/SimplenoteAppDelegate+Swift.swift:10]: I forgot about this little bit of API.  Nice :)

File: Simplenote/SplitViewController.swift [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    override func loadView() {
+        let splitView = SplitView()
+        splitView.isVertical = true
+        self.splitView = splitView
+        self.view = splitView
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        splitView.autosaveName = "Please Save Me!"
```
Comment (aerych) [inline Simplenote/SplitViewController.swift:27]: 😆 I had to google to see what this was about.  Is that from [this](https://stackoverflow.com/questions/16587058/nssplitview-auto-saving-divider-positions-doesnt-work-with-auto-layout-enable)?  Should we add a citation, or maybe leave a comment about the glitch?

Comment (jleandroperez) [reply Simplenote/SplitViewController.swift:27]: Wow!! Actually, I was aware of this problem (divider positions not being saved). We had a really ugly fix, in the past, which is being removed in this very same PR.

The string matching is **an astronomical coincidence** (I was listening to an obscure song when this was written!!).

I'll add a comment, thank you!!!

**Comment by jleandroperez** [comment]
Thanks **A LOT** for testing this one @aerych + @emilylaguna !!!

**Comment by jleandroperez** [comment]
P.s.: Adding a note on the main issue, to improve the divider(s) refresh mechanism!

**Pull request merged**